README.md
# multiutillib

A flutter package to help with utilities, animations and some pre-built widgets.
Some widgets are with default values for direct use and you can still customize according to your needs.

This package used following packages:
1. [intl](https://pub.dev/packages/intl)
2. [shimmer](https://pub.dev/packages/shimmer)
3. [device_info](https://pub.dev/packages/device_info)
4. [url_launcher](https://pub.dev/packages/url_launcher)
5. [package_info](https://pub.dev/packages/package_info)
6. [simple_animations](https://pub.dev/packages/simple_animations)
7. [connectivity_plus](https://pub.dev/packages/connectivity_plus)
8. [loading_animation_widget](https://pub.dev/packages/loading_animation_widget)

## Examples

## FadeAnimation

This widget will help a child widget to perform a fade animation.
[FadeAnimation] (lib/animations/fade_animation.dart) requires two parameters delay and child.
The delay is animation delay to be performed.

Example of FadeAnimation:

```dart
FadeAnimation(
    delay: 1,
    child: Container(),
),
```

## SlideAnimation

This widget will help a child widget to perform a slide animation from Top, Left, Right or Bottom direction.
[SlideAnimation] (lib/animations/slide_animation.dart) required 5 parameters child, itemCount, position, slideDirection and animationController.
This widget will help in animating items of list view.

There is enum to help choose direction of slide animation.

Example of SlideAnimation:

```dart
SlideAnimation(
    child: child,
    position: position,
    itemCount: itemCount,
    slideDirection: SlideDirection.fromRight,
    animationController: animationController,
),
```

## OffsetAnimation

This widget will help a child widget to perform offset animation

Example of OffsetAnimation:

```dart
OffsetAnimation(
  end: 20, // the end parameter is optional
  animationController: _animationController, // required parameter
  widget: YourChildWidget, // required parameter
),
```

## DateTimeUtils

This is a utility class which will help in getting current date in format 'yyyy-MM-dd HH:mm:ss' or date format passed.
This class will also help in formatting date in any valid format like 'dd-MM-yyyy' or 'dd/MM/yyyy' or any other you want.

Example of DateTimeUtils:

```dart
debugPrint('current date is: ${getCurrentDate()}');
```

There is extension methods which can be used to display date, time in provided or default format

Example:

```dart
// Convert time String to Duration Object
'01:01:01'.toDuration;

// Convert Date String to Date Time Object
'20210226'.toDateTime;
'2021-02-26'.toDateTime;
'20210226 140652.55'.toDateTime
'2021-02-26 14:06:52.55'.toDateTime

// Convert Date Time to Date Time String for display
DateTime.now().formatDateTime(newDateTimeFormat: 'dd-MM-yyyy hh:mm a');

// Convert Date Time String to Date Time String for display
DateTime.now().formatDateTime();

// getCurrentDate() returns date time string
getCurrentDate().formatDateTime(newDateTimeFormat: 'dd-MM-yyyy');

// Convert Time of Day to Time String for display
TimeOfDay.now().formatTime();

// Convert Time of Day String to Time String for display
'14:39'.formatTime();

// Convert Time String to Time Of Day Object
'14:39'.toTimeOfDay

// Convert Date Time to Time of Day Object
DateTime.now().toTimeOfDay
```

## Utils

This is a utility class which will help in getting device name, device id, managing null values in strings.

```dart
1. getDeviceName() // - returns future string
2. getDeviceId() // - returns future string
3. getSingleDigitRandomNumber() // - returns single digit randomly
4. getRandomNumber(min: MinValue, max: MaxValue) // - returns randomly generated number between min and max value
5. getRandomMobileNumber() // - returns randomly generated mobile number
6. getCharFromString(stringToExtract) // - returns only characters from string passed
7. getNumbersFromString(stringToExtract) // - returns only digits from string passed
8. getAppVersion() // - return current version of the app
```

## Extensions

Some of the Utils method can now be used as extensions.

Example:

```dart
// Checking if a string or int or double value is numeric or not
'123'.isNumeric => returns true
'String'.isNumeric => returns false

// Checking if a string is null or empty
strToCheck.isNullOrEmpty

// for removing last character from a string
stringToRemoveLastChat.trimLastChar

// Converting a string to two digits
Text('To Two Digit: ${'9'.toTwoDigits}'),

// Converting a string to three digits
Text('To Three Digit: ${'9'.toThreeDigits}'),

// Converting a string to a fixed length
Text('To Length of String: ${'9'.toLengthOfString(5)}'),

// Formatting numbers with extensions
Text('Format Number: ' + '1010.869'.formatNumber()),
Text('Format Number to Compact: ' + '88944444'.formatNumberToCompact), // return 1K, 2k, 2M

Text('Replace Null With Empty: ' + 'null'.replaceNullWithEmpty),
Text('Replace True or False: ' + 'true'.replaceTrueOrFalse.toString()),
Text('Replace Null With Zero: ' + 'null'.replaceNullWithZero.toString()),
Text('Replace Null with Double: ' + 'null'.replaceNullWithDouble.toString()),
Text('Convert Duration to Time String: ' + '16:35:34'.toDuration.toTimeString(newTimeString: 'mm:ss')),
```

## ShowConfirmationDialog

This method will show a confirmation dialog box with two button to ask for confirmation.

Example:

```dart
showConfirmationDialog(
    BuildContext context,
    Widget transitionAnimation,
    String negativeBtnText 'No',
    String positiveBtnText: 'Yes',
    bool barrierDismissible: false,
    Color dividerColor: Colors.blue,
    Color positiveBtnColor: Colors.blue,
    Color negativeBtnColor: Colors.blueAccent,
    TextAlign descTextAlign: TextAlign.center,
    TextAlign titleTextAlign: TextAlign.center,
    Duration transitionDuration: const Duration(milliseconds: 400),
    DialogAnimationType dialogAnimationType: DialogAnimationType.grow,
    TextStyle descStyle: const TextStyle(fontSize: 16, letterSpacing: 0.27, color: Colors.black),
    TextStyle titleStyle: const TextStyle(fontSize: 18, letterSpacing: 0.27, fontWeight: FontWeight.bold),
    TextStyle positiveBtnStyle: const TextStyle(fontSize: 18, letterSpacing: 0.27, fontWeight: FontWeight.w400),
    TextStyle negativeBtnStyle: const TextStyle(fontSize: 18, letterSpacing: 0.27, fontWeight: FontWeight.w400),
    required String title,
    required String description,
    required VoidCallback onPositivePressed,
);
```

## ShowCustomDialog

This method will show a dialog box with single button to show warning or info or success or error.

Example:

```dart
showCustomDialog(
  BuildContext context,
  String btnText: 'OK',
  Color btnColor: Colors.blue,
  bool barrierDismissible: false,
  Color dividerColor: Colors.blue,
  TextAlign descTextAlign: TextAlign.center,
  TextAlign titleTextAlign: TextAlign.center,
  Duration transitionDuration: const Duration(milliseconds: 400),
  DialogAnimationType dialogAnimationType: DialogAnimationType.grow,
  TextStyle descStyle: const TextStyle(fontSize: 16, letterSpacing: 0.27, color: Colors.black),
  TextStyle btnStyle: const TextStyle(fontSize: 18, letterSpacing: 0.27, fontWeight: FontWeight.w400),
  TextStyle titleStyle: const TextStyle(fontSize: 18, letterSpacing: 0.27, fontWeight: FontWeight.bold),
  required String title,
  required String description,
);
```

## CustomDateRangePicker

This class wil help display custom date range picker dialog.

Example: 

```dart
showCustomDateRangePicker(
  required BuildContext context,
  required Function onCancelClick,
  String applyButtonText: 'Apply',
  String cancelButtonText: 'Cancel',
  Color leftArrowColor: Colors.blue,
  Color rightArrowColor: Colors.blue,
  Color applyButtonColor: Colors.blue,
  Color cancelButtonColor: Colors.red,
  Color weekDaysTextColor: Colors.blue,
  Color selectedRangeColor: Colors.blue,
  Color monthYearTextColor: Colors.black,
  required Function(DateTime startDate, DateTime endDate) onApplyClick,
  TextStyle applyButtonTextStyle: const TextStyle(fontSize: 18, color: Colors.white, fontWeight: FontWeight.bold),
  TextStyle cancelButtonTextStyle: const TextStyle(fontSize: 18, color: Colors.white, fontWeight: FontWeight.bold),
  DateTime minimumDate,
  DateTime maximumDate,
  DateTime initialEndDate,
  DateTime initialStartDate,
  bool barrierDismissible: false,
);
```

## CustomDatePicker

This class wil help display custom date picker dialog.

Example: 

```dart
showCustomDatePicker(
  required BuildContext context,
  required Function onCancelClick,
  String applyButtonText: 'Apply',
  String cancelButtonText: 'Cancel',
  Color leftArrowColor: Colors.blue,
  Color rightArrowColor: Colors.blue,
  Color applyButtonColor: Colors.blue,
  Color cancelButtonColor: Colors.red,
  Color weekDaysTextColor: Colors.blue,
  Color selectedDateColor: Colors.blue,
  Color monthYearTextColor: Colors.black,
  required Function(DateTime dateTime) onApplyClick,
  TextStyle applyButtonTextStyle: const TextStyle(fontSize: 18, color: Colors.white, fontWeight: FontWeight.bold),
  TextStyle cancelButtonTextStyle: const TextStyle(fontSize: 18, color: Colors.white, fontWeight: FontWeight.bold),
  DateTime minimumDate,
  DateTime maximumDate,
  DateTime initialStartDate,
  bool barrierDismissible: false,
);
```

## ProgressDialog

This class will help in showing and hiding progress dialog with custom loading widget.

Example:

```dart
// For Showing Progress Dialog
await showProgressDialog(context);

// For hiding Progress Dialog
await hideProgressDialog();
```

## ConnectivityBuilder
This is widget to show connectivity message like "Internet Connection not available".

Example:

```dart
ConnectivityBuilder(
    Key? key,
    required builder,
    gradient,
    offlineWidget,
    bgColor: Colors.red,
    position: Position.bottom,
    disableInteraction: false,
    alignment: Alignment.center,
    message: kInternetNotAvailable,
    textStyle: const TextStyle(fontSize: 14, color: Colors.white),
),
```

## DefaultButton

This is a button widget. See example for using this widget

Example:

```dart
DefaultButton(
    isEnabled: true,
    isUpperCase: true,
    margin: const EdgeInsets.only(top: 20),
    required String text,
    required Function onPressed,
),
```

## HyperLinkText

This is a widget to show a text and launch an url on click of the text.

Example:

```dart
HyperLinkText(url: 'http://www.google.com', text: 'Open Google'),
```

## LoadingWidget

This widget will show a loading shimmer effect.

Example:

```dart
LoadingWidget(
    itemCount: 6,
    borderRadius: 12,
    highlightColor: Colors.white,
    baseColor: const Color(0xFFE0E0E0),
),
```

## MaterialCard

This widget is useful in showing card view. You can use it directly or customize according to your needs.
You can pass either borderRadius parameter value or borderRadiusGeometry parameter value.

Example:

```dart
MaterialCard(
    required Widget child,
    onTap,
    borderRadius,
    elevation: 4,
    borderRadiusGeometry,
    color: Colors.white,
    padding: const EdgeInsets.all(12),
    margin: const EdgeInsets.only(top: 12),
),
```

## OTPTextField

This widget is useful to take input for OTP while using OTP based login.

Example:

```dart
OTPTextField(
    required int noOfOtpFields,
    required ValueChanged<String> onCompleted,
    borderWidth: 2,
    isEnabled: true,
    autoFocus: true,
    autoCorrect: false,
    borderColor: Colors.blue,
    cursorColor: Colors.blue,
    textAlign: TextAlign.center,
    margin: const EdgeInsets.only(top: 20, left: 20, right: 20),
    textStyle: const TextStyle(fontSize: 16, letterSpacing: 0.27, color: Colors.black),
),
```

## RichTextWidget

This widget is helpful in showing caption and description value in single line or caption on top and description in new line.

Example:

```dart
RichTextWidget(
    required String caption,
    required String description,
    isDescNewLine: false,
    captionStyle: const TextStyle(fontSize: 14, color: Colors.black, fontWeight: FontWeight.normal),
    descriptionStyle: const TextStyle(fontSize: 14, color: Colors.black, fontWeight: FontWeight.normal),
),
```

## CustomToggleButton

This widget will display toggle switch with text.

Example:

```dart
CustomToggleButton({
    textOn,
    textOff,
    onChanged,
    width: 200,
    height: 45,
    borderRadius: 50,
    transitionTime: 300,
    activeTextColor: Colors.black,
    activeSwitchColor: Colors.white,
    inactiveTextColor: Colors.white,
    inactiveSwitchColor: Colors.blue,
});
```

## DecimalInputFormatter

This is an [TextInputFormatter] for [TextField] for allowing to enter only allowed decimal range

Example:
```dart
// [DecimalInputFormatter] will allow only entered decimal range for the string in text field
TextFormField(
  inputFormatters: [DecimalInputFormatter(decimalRange: 2)],
  decoration: InputDecoration(labelText: 'Enter Decimal values'),
),
```

## Validators

This class has validation method for validating string

1. Email Validator - This will check and validate if the entered string is a valid email id

Example:

```dart
TextFormField(
  // this method will validate for email id
  validator: emailValidator,
  decoration: InputDecoration(labelText: 'Enter Email Id'),
),
```


CHANGELOG.md
# [1.3.0] 16-09-2023
* Updated all the dependencies to latest version.

# [1.2.2] 30-10-2022
* Updated all the dependencies to latest version.

# [1.2.1] 22-08-2022
* Added String to Color conversion extension

# [1.2.0] 17-06-2022
* Added Debounce Util.
* Changes in Progress dialog loader.
* Added loading animation widget.

# [1.1.9] 19-01-2022
* Added Connectivity Builder Widget.
* Removed Connectivity Layout Widget.
* Added [isNullOrEmpty] string extension.
* Added [trimLastChat] string extension.

# [1.1.8] 11-01-2022
* Updated all the dependencies.
* Deprecating Connectivity Layout Widget.

# [1.1.7] 28-10-2021
* Updated all the dependencies.
* Updated dart version to 2.14.0.

# [1.1.6] 23-09-2021
* Removed src folder and moved all files outside src.

# [1.1.5] 23-09-2021
* Updated other packages used in this package.
* Added border radius field to Loading widget.

# [1.1.4] 25-08-2021
* Resolved bug in ui helper's horizontal spacing.
* Now you can add text style to rich text widget individually.

# [1.1.3] 30-07-2021
* Resolved bug in connectivity layout.

# [1.1.2] 16-07-2021
* Optimized code after adding analysis options.
* Added UI Helpers and Custom App Bar.

## [1.1.1] 10-06-2021
* Bug fixes after migrating to null safety.

## [1.1.0] 31-05-2021
* Added new extension for formatting number in compact view.
* Migrated to Null Safety.

## [1.0.6] - 22-05-2021
* Fixed bug where margin was not applied to Default Button.
* 2. Added parameter to increase height of Default Button.

## [1.0.5] - 25-03-2021
* Updated supporting packages version.

## [1.0.4] - 15-03-2021
* Added Custom Toggle Button.
* Added Custom Date Picker Dialog.
* Added Custom Date Range Picker Dialog.

## [1.0.3] - 12-03-2021
* Added shadow color option for MaterialCard widget.
* Code modification and removal of unwanted codes.

## [1.0.2] - 05-03-2021
* Added new extensions for ease of use for converting.

## [1.0.1] - 26-02-2021
* Added extensions for ease of use for converting.

## [1.0.0] - 26-02-2021
* Major Changes in package.
* Deprecated related methods used in extension.
* Added extensions for ease of use for converting.

## [0.0.6] - 25-02-2021
* Fixed bug in OTPTextField widget where entered otp was not available
* Updated DefaultButton to use ElevatedButton

## [0.0.5] - 24-02-2021
* Added new method for getting current app version
* Updated dependent package version.

## [0.0.4] - 21-01-2021
* Added comments for use.
* Added new utility methods.

## [0.0.3] - 29-12-2020
* Added formatting to documentation.

## [0.0.2] - 29-12-2020
* Added new method in Utils class for formatting number in given pattern.
* Added documentation for showing how to use the widgets and utilities.

## [0.0.1] - 26-12-2020
* Initial Release


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:intl/intl.dart';
import 'package:loading_animation_widget/loading_animation_widget.dart';
import 'package:multiutillib/multiutillib.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: const MyHomePage(),
      title: 'multiutillib Example',
      debugShowCheckedModeBanner: false,
      theme: ThemeData(primarySwatch: Colors.blue, visualDensity: VisualDensity.adaptivePlatformDensity),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  MyHomePageState createState() => MyHomePageState();
}

class MyHomePageState extends State<MyHomePage> with SingleTickerProviderStateMixin {
  late final AnimationController _animationController;
  final TextEditingController _textEditingController = TextEditingController();

  String _isValid = '* Email Id is required.';

  @override
  void initState() {
    super.initState();

    _animationController = AnimationController(vsync: this, duration: const Duration(seconds: 1));
    _animationController.forward();

    _textEditingController.addListener(() {
      final isEmailValid = emailValidator(_textEditingController.value.text);

      if (isEmailValid != _isValid) {
        setState(() => _isValid = isEmailValid!);
      }
    });

    // _showProgressDialog();
  }

  /*_showProgressDialog() async {
    WidgetsBinding.instance.addPostFrameCallback((timeStamp) async {
      await showInkDropProgressDialog(context);

      await Future.delayed(const Duration(seconds: 3));

      await hideProgressDialog();
    });
  }*/

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.grey[300],
      appBar: AppBar(title: const Text('Multi Util Lib')),
      body: ConnectivityBuilder(
        builder: (isOnline) {
          return MaterialCard(
            borderRadius: 15,
            // color: Colors.grey[300],
            margin: const EdgeInsets.all(15),
            child: SingleChildScrollView(
              child: Column(
                children: <Widget>[
                  const SizedBox(height: 10),
                  CustomToggleButton(
                    textOn: 'On',
                    textOff: 'Off',
                    onChanged: (value) => debugPrint(value),
                  ),
                  const SizedBox(height: 10),
                  FadeAnimation(
                    delay: 1.5,
                    child: Container(height: 100, color: Colors.black12, width: double.infinity),
                  ),
                  const SizedBox(height: 10),
                  LoadingAnimationWidget.inkDrop(color: Colors.blue, size: 36),
                  const SizedBox(height: 10),
                  Text('Get Current Date: ${getCurrentDate()}'),
                  const SizedBox(height: 10),
                  Text(
                    '${'Get Current Date in $kDateDisplayFormat Format: \n'}${getCurrentDate(newDateTimeFormat: kDateDisplayFormat)}',
                    textAlign: TextAlign.center,
                  ),
                  const SizedBox(height: 10),
                  Text(
                    '${'Get Current Date in $kFullMonthYearFormat Format: \n'}${getCurrentDate(newDateTimeFormat: kFullMonthYearFormat)}',
                    textAlign: TextAlign.center,
                  ),
                  const SizedBox(height: 10),
                  Text(
                    '${'Get Current Date in $kFullDateDisplayFormat Format: \n'}${getCurrentDate(newDateTimeFormat: kFullDateDisplayFormat)}',
                    textAlign: TextAlign.center,
                  ),
                  const SizedBox(height: 10),
                  Text(
                    'Format Date Time in dd-MM-yyyy format: \n${getCurrentDate().formatDateTime(newDateTimeFormat: 'dd-MM-yyyy')}',
                    textAlign: TextAlign.center,
                  ),
                  const SizedBox(height: 10),
                  Text(
                    'Formatting Date Time Object in dd-MM-yyyy format: \n${DateTime.now().formatDateTime(newDateTimeFormat: 'dd-MM-yyyy HH:mm')}',
                    textAlign: TextAlign.center,
                  ),
                  const SizedBox(height: 10),
                  Text(
                    'Convert to Date Time Object: \n${'2021-02-26 14:06:52.55'.toDateTime}',
                    textAlign: TextAlign.center,
                  ),
                  const SizedBox(height: 10),
                  Text(
                    'Convert Time of Day to String: \n${TimeOfDay.now().formatTime()}',
                    textAlign: TextAlign.center,
                  ),
                  const SizedBox(height: 10),
                  Text(
                    'Convert Time of Day String to String: \n${'14:39'.formatTime()}',
                    textAlign: TextAlign.center,
                  ),
                  const SizedBox(height: 10),
                  Text(
                    'Convert Time String to Time of Day: \n${'14:39'.toTimeOfDay}',
                    textAlign: TextAlign.center,
                  ),
                  const SizedBox(height: 10),
                  Text(
                    'Convert DateTime to Time of Day: \n${DateTime.now().toTimeOfDay}',
                    textAlign: TextAlign.center,
                  ),
                  const SizedBox(height: 10),
                  Text(
                    "Convert DateTime to String: \n${DateTime.now().formatDateTime()}",
                    textAlign: TextAlign.center,
                  ),
                  const SizedBox(height: 10),
                  Text('Replace Null With Empty: ${"null".replaceNullWithEmpty}'),
                  const SizedBox(height: 10),
                  Text('Replace Null With Zero: ${"null".replaceNullWithZero}'),
                  const SizedBox(height: 10),
                  Text('Replace True or False: ${"true".replaceTrueOrFalse}'),
                  const SizedBox(height: 10),
                  Text('Replace Null with Double: ${"null".replaceNullWithDouble}'),
                  const SizedBox(height: 10),
                  Text('Format Number: ${"1010.869".formatNumber()}'),
                  const SizedBox(height: 10),
                  Text('Format Number to Compact: ${'88944444'.formatNumberToCompact}'),
                  const SizedBox(height: 10),
                  Text('Is Numeric: ${"1".isNumeric}'),
                  const SizedBox(height: 10),
                  Text('To Duration: ${'01:01:01'.toDuration}'),
                  const SizedBox(height: 10),
                  Text('From Duration: ${'16:35:34'.toDuration.toTimeString(newTimeString: 'mm:ss')}'),
                  const SizedBox(height: 10),
                  Text('To Two Digit: ${"9".toTwoDigits}'),
                  const SizedBox(height: 10),
                  Text('To Three Digit: ${"9".toThreeDigits}'),
                  const SizedBox(height: 10),
                  Text('To Length of String: ${'9'.toLengthOfString(5)}'),
                  const SizedBox(height: 10),
                  // [DecimalInputFormatter] will allow only entered decimal range for the string in text field
                  TextFormField(
                    inputFormatters: [DecimalInputFormatter(decimalRange: 2)],
                    decoration: const InputDecoration(labelText: 'Enter Decimal values'),
                  ),
                  const SizedBox(height: 10),
                  TextFormField(
                    // this method will validate for email id
                    validator: emailValidator,
                    controller: _textEditingController,
                    keyboardType: TextInputType.emailAddress,
                    decoration: InputDecoration(errorText: _isValid, labelText: 'Enter Email Id'),
                  ),
                  const SizedBox(height: 10),
                  FutureBuilder(
                    future: getDeviceName(),
                    builder: (context, snapshot) {
                      var formattedNumber = NumberFormat.compactCurrency(
                        symbol: '',
                        locale: 'en_IN',
                        decimalDigits: 2,
                      ).format(100000);

                      debugPrint('Formatted Number is $formattedNumber');
                      debugPrint('snapshot data is: ${snapshot.data}');

                      if (snapshot.hasData) return Text(snapshot.data!);

                      return const SizedBox.shrink();
                    },
                  ),
                  const SizedBox(height: 10),
                  FutureBuilder(
                    future: getDeviceId(),
                    builder: (context, snapshot) {
                      debugPrint('snapshot data is: ${snapshot.data}');

                      if (snapshot.hasData) return Text(snapshot.data!);

                      return const SizedBox.shrink();
                    },
                  ),
                  const SizedBox(height: 10),
                  const HyperLinkText(url: 'http://www.google.com', text: 'Open Google'),
                  DefaultButton(
                    text: 'Show Date Range Picker',
                    margin: const EdgeInsets.only(top: 10),
                    onPressed: () {
                      showCustomDateRangePicker(
                        context: context,
                        onCancelClick: () {},
                        onApplyClick: (startDate, enDate) {},
                      );
                    },
                  ),
                  DefaultButton(
                    text: 'Show Date Picker',
                    margin: const EdgeInsets.only(top: 10),
                    onPressed: () {
                      showCustomDatePicker(
                        context: context,
                        onCancelClick: () {},
                        onApplyClick: (selectedDate) {},
                      );
                    },
                  ),
                  DefaultButton(
                    text: 'Show Custom Dialog',
                    margin: const EdgeInsets.only(top: 10),
                    onPressed: () {
                      showCustomDialog(
                        context,
                        title: 'Custom Dialog',
                        btnColor: Colors.green,
                        dialogAnimationType: DialogAnimationType.fromBottom,
                        description: 'This is a test to show custom dialog',
                      );
                    },
                  ),
                  DefaultButton(
                    text: 'Show Confirmation Dialog',
                    margin: const EdgeInsets.only(top: 10),
                    onPressed: () {
                      showConfirmationDialog(
                        context,
                        dividerColor: Colors.black,
                        title: 'Confirmation Dialog',
                        positiveBtnColor: Colors.green,
                        negativeBtnColor: Colors.greenAccent,
                        onPositivePressed: () => Navigator.pop(context),
                        dialogAnimationType: DialogAnimationType.fromLeft,
                        description: 'This is a test to show confirmation dialog',
                      );
                    },
                  ),
                  DefaultButton(
                    text: 'Show Progress Dialog',
                    margin: const EdgeInsets.only(top: 10),
                    onPressed: () async {
                      // calling show progress dialog method
                      await showProgressDialog(
                        context, /*progressWidget: const CircularProgressIndicator()*/
                      );

                      await Future.delayed(const Duration(seconds: 3));

                      await hideProgressDialog();
                    },
                  ),
                  const SizedBox(height: 10),
                  const LoadingWidget(itemCount: 1),
                  OTPTextField(
                    autoFocus: false,
                    noOfOtpFields: 6,
                    onCompleted: (enteredOtp) => debugPrint('entered otp is: $enteredOtp'),
                  ),
                  const SizedBox(height: 10),
                  const RichTextWidget(
                    caption: 'Rich',
                    description: 'Text',
                    captionStyle: TextStyle(fontSize: 20, color: Colors.black),
                    descriptionStyle: TextStyle(fontSize: 24, color: Colors.black),
                  ),
                ],
              ),
            ),
          );
        },
      ),
    );
  }

  @override
  void dispose() {
    _animationController.dispose();
    _textEditingController.dispose();

    super.dispose();
  }
}
